### ISSUE TITLE
Support of compressed PLY

### ISSUE THREAD

**Issue by alexey7783** [aras-p/UnityGaussianSplatting#141]
Hi guys,

is support of compressed PLY on the roadmap? Uncompressed are just so gigantic and not fun to work with...

Thanks and regards

**Comment by aras-p** [comment]
@alexey7783 the whole point of this repository is that you import raw uncompressed PLY, and then _within unity_, your 3DGS data gets compressed (depending on your settings). So once it is in Unity, the size of original PLY does not matter anymore.

That said, what do you mean by "compressed PLY"? The variant used by SuperSplat (which also drops spherical harmonics and thus loses lighting fidelity), or something else?

**Comment by alexey7783** [comment]
Hi @aras-p Thanks for the reply. The issue currently is, that Unity seems to have a 2GB threshold for the uncompressed PLY. We want to use complete environments which can be even larger. 

And yes I mean the version of SuperSplat. What about the .splat format? Does this still contain the SH?

**Comment by aras-p** [comment]
@alexey7783 ah, the 2GB / "around 8 million splats" limitation is not directly related to file size, but due to other decisions made in this project. Just using "some sort of smaller PLY format" would not address or fix them -- as in, if someone would contribute support for SuperSplat compressed format, you'd _still be limited_ to around 8 million splats in one object.

As far as I know, .splat format (if you mean https://github.com/antimatter15/splat format and not something else) also does not have spherical harmonics.

**Comment by alexey7783** [comment]
@aras-p I learn so much from you, thanks :-)

**Comment by unitycoder** [comment]
just for info,
another (quantized+compressed) format coming up here
https://scaniverse.com/news/spz-gaussian-splat-open-source-file-format (just open sourced)

**Comment by aras-p** [comment]
@unitycoder yeah just saw it. From the first look, it is worse compression ratio than data storage formats here. It is no longer "PLY" however but an entirely different format.

**Comment by alexey7783** [comment]
@aras-p Do they maintain the spherical harmonics? The scaniverse app ist just a toy for now, because the resulting splats are very very very basic and ugly from upclose. They are computed on the mobile device. You can refine a few times, but they are far inferior to splats created with the PC tools.

**Comment by aras-p** [comment]
@alexey7783 I know nothing about quality or lack of it of Scaniverse, just writing my opinion on their .SPZ format that I have spent a total of 3 minutes reading about :)
